### PR TITLE
add terraform linter

### PR DESCRIPTION
**Pull request by morozovalekseywot**

### CODE REVIEW

File: .tflint.hcl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+rule "terraform_required_version" {
+  enabled = false
+}
+
+rule "terraform_required_providers" {
+  enabled = false
+}
+
+rule "terraform_unused_declarations" {
```
Comment (paaanic) [inline .tflint.hcl:18]: Why is it disabled?

File: .github/workflows/tf-lint.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: TF CI
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:1]:
```suggestion
name: Check Examples
```

File: .github/workflows/tf-lint.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:8]: Specify paths for examples/linting config/makefile.

File: .github/workflows/tf-lint.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+jobs:
+  fmt:
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:10]:
```suggestion
  check-terraform-fmt:
```

File: .github/workflows/tf-lint.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Terraform with specified version on the runner
+        uses: hashicorp/setup-terraform@v3
+
+      - name: Terraform format
+        id: fmt
+        run: terraform fmt -check -recursive
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:24]: Lets run only on examples directory.

File: .github/workflows/tf-lint.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  lint:
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:25]:
```suggestion
  lint-terraform:
```

File: GNUmakefile [hunk 7]
```diff
@@ -90,4 +90,13 @@ lint:
...
 
+tflint:
+	tflint --recursive -f compact --config="$(CURDIR)/.tflint.hcl" --fix
+
+tflint_check:
+	tflint --recursive -f compact --config="$(CURDIR)/.tflint.hcl"
+
+tf_fmt:
+	terraform fmt -recursive
+
```
Comment (paaanic) [inline GNUmakefile:101]:
```
tflint_fix_examples:
 # fix examples linting
tffmt_examples:
 # format examples
check_examples:
 # linting + formatting checks on examples directory
```

**Review by paaanic — CHANGES_REQUESTED** [review]
LGTM overall, only several minor changes.

File: .github/workflows/tf-lint.yml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Terraform with specified version on the runner
+        uses: hashicorp/setup-terraform@v3
+
+      - name: Terraform format
```
Comment (paaanic) [inline .github/workflows/tf-lint.yml:22]:
```suggestion
      - name: Check Terraform formatting
```

File: .tflint.hcl (hunk 1 above)
Comment (morozovalekseywot) [reply .tflint.hcl:18]: some of data sources in our examples is unused (has no output)

File: GNUmakefile (hunk 7 above)
Comment (morozovalekseywot) [reply GNUmakefile:101]: may be shorter?
- tflint_fix
- tffmt
- check_examples

If you want to do formatting, then it takes a long time to type such commands. And we have terraform files anyway only in examples

Comment (paaanic) [reply GNUmakefile]: Setup bash autocompletion. :)

**Review by paaanic — APPROVED** [review]

**Pull request merged**